ls1intum/Artemis
# `Communication`: Add all participants to default channels automatically

### Checklist
#### General


#### Server

### Motivation and Context


#6487 creates 4 default channels when a course is created, but participants are not added automatically yet.

### Description

Upon course creation, the 4 default channels are created and all students, tutors, and instructors of the course are added to them.
Additionally, if a user is added to a group (which potentially belongs to multiple courses) they join all the default channels of all the courses the user is added to.

### Steps for Testing

Prerequisites:
- 2 Instructors
- 1 Student
- 1 Tutor
- (1 Editor)

1. Log in to Artemis as one instructor
2. Navigate to Course Administration
3. Create a course. Enable messaging
4. Add the other users (1 student, 1 tutor, 1 additional instructor, (1 editor)) to the course
5. Check if the other users are automatically added to all 4 default channels
6. Create another course with the same student/tutor/instructor(/editor) groups
7. Check if all users are automatically added to the 4 default channels of the new course

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage




#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [DefaultChannelType.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.enumeration/DefaultChannelType.html) | 100% | ✅ |
| [CourseRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/CourseRepository.html) | 81% | ✅ |
| [ChannelService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ChannelService.html) | 95% | ✅ |
| [ConversationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ConversationService.html) | 96% | ✅ |
| [CourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/CourseResource.html) | 94% | ✅ |
| [AdminCourseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.admin/AdminCourseResource.html) | 100% | ✅ |
| [ChannelResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4788-22/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.metis.conversation/ChannelResource.html) | 91% | ✅ |
